Trigger markdown on any character and change markdown implementation

mxstbr opened the pull request:
**DEMO:  https://public-myglckweth.now.sh**

This PR makes it so that markdown is created when pressing the closing character of the markdown instance, not when pressing space which is a much nicer experience for the users.

Due to that, I had to change the markdown implementation to only allow a single way of doing things that doesn't conflict with each other:

- **bold** will be triggered by `*bold*`
- _italic_ will be triggered by `_italic_`
- ~~strikethrough~~ will be triggered by `~strikethrough~`

Nothing else should have changed.

mxstbr committed:
Trigger *bold* and _italic_

mxstbr committed:
Trigger markdown on any character

This makes it so typing `*` after `*some text` immediately makes it bold
without having to press space after!

mxstbr committed:
Change strikethrough to ~text~

mxstbr committed:
Update readme

superbryntendo commented:
This automatically inserts a space after the styles convert, which isn't ideal. Otherwise, seems good!

We'll also need to update the cheatsheet next to the chat input...

mxstbr commented:
> We'll also need to update the cheatsheet next to the chat input...

Yep, totally, good catch. Will do that when I published the new version of the plugin and update it in the main codebase!

mxstbr commented:
> This automatically inserts a space after the styles convert, which isn't ideal. 

I thought about this but it feels very awkward without the space—let me have another play with it.

mxstbr committed:
Disallow users from typing two spaces

@uberbryn I couldn't make it so we don't automatically type the space,
so I've just prevented users from double typing spaces in general.

mxstbr commented:
@uberbryn I have to insert the space after, otherwise I don't know how to make you exit the style :confused: Demo of the bug that ensues without the space:

![](https://i.imgur.com/MHvNDpG.gif) 

Since I don't know how to fix that I've worked around the issue by disallowing users from typing two spaces in succession, which should mean that even if they hit space after inserting the style they don't end up with a double space.

New demo: https://public-myglckweth.now.sh (also updated in the OP)

mxstbr commented:
I'm going to ship this for now and we can iterate on it later!
